src/main/java/com/crowdin/cli/commands/actions/StatusAction.java

```
@@ -78,20 +83,38 @@ public void act(Outputter out, ProjectProperties pb, ProjectClient client) {
                 }
             });
         } else {
+            Stream<LanguageProgress> languageProgressStream = progresses.stream();
+
             if (showTranslated && showApproved) {
                 out.println(RESOURCE_BUNDLE.getString("message.translation"));
             }
             if (showTranslated) {
+                throwExceptionIfIncomplete(languageProgressStream.filter(p -> p.getTranslationProgress() <= 100),failIfIncomplete);
+
                 progresses.forEach(pr -> out.println(String.format(RESOURCE_BUNDLE.getString("message.item_list_with_percents"),
                     pr.getLanguageId(), pr.getTranslationProgress())));
             }
             if (showTranslated && showApproved) {
                 out.println(RESOURCE_BUNDLE.getString("message.approval"));
             }
             if (showApproved) {
+                throwExceptionIfIncomplete(languageProgressStream.filter(p -> p.getApprovalProgress() <= 100),failIfIncomplete);
                 progresses.forEach(pr -> out.println(String.format(RESOURCE_BUNDLE.getString("message.item_list_with_percents"),
                     pr.getLanguageId(), pr.getApprovalProgress())));
             }
+            if (failIfIncomplete) {
+                throwExceptionIfIncomplete(languageProgressStream.filter(p -> p.getApprovalProgress() <= 100),failIfIncomplete);
+            }
+        }
+    }
+
+    private Consumer<? super LanguageProgress> throwException(String msg) {
```

do we still need this method? Can we throw the exception in the method below?

---

I am using this method for wrapper exception throwing